Add GitHub Action for PR-level testing

iampearlep opened the pull request:
This PR adds a new GitHub Action that triggers the PR level to test if everything works without issues. The action runs yarn lint, yarn type-check, and yarn build to ensure the code is correct and builds successfully.

vault-developer commented on pr-test.yml:
```diff
@@ -0,0 +1,41 @@
+
+      # Install dependencies
+      - name: Install dependencies
+        run: yarn install
```
Could you please check if it is working correctly with yarn v4?
My concern is that we might need to enable corepack explicitly.

iampearlep replied:
Noted, I will work on these changes

vault-developer reviewed (changes requested):
Hey @iampearlep thank you for the PR 👍 
I'm afraid this action won't work due to the reasons mentioned in the comments.
May I ask you to address it please and test github action before sending PR to review?

iampearlep commented:
I have made changes and tested and it works fine, thank you.

vault-developer commented:
Hey @iampearlep thank you for addressing comments.
I tested GA and it's not working due to the corepack, the issue I mentioned earlier in the discussion.
https://github.com/vault-developer/event-loop-explorer/actions/runs/11191445539/job/31116863552?pr=11

May I ask how did you test it?
I also checked your fork and it looks like it was never triggered there:
https://github.com/iampearlep/event-loop-explorer/actions/workflows/pr-check.yml

iampearlep commented:
> Hey @iampearlep thank you for addressing comments. I tested GA and it's not working due to the corepack, the issue I mentioned earlier in the discussion. https://github.com/vault-developer/event-loop-explorer/actions/runs/11191445539/job/31116863552?pr=11
> 
> May I ask how did you test it? I also checked your fork and it looks like it was never triggered there: https://github.com/iampearlep/event-loop-explorer/actions/workflows/pr-check.yml

Thank you for your feedback, I apologize for the error in the workflow. I initially tested the pr-check action on another repository and now realize I should have tested it in this repository. I will make the necessary adjustments to the workflow and test it again here.

iampearlep commented:
Hello @vault-developer 
Thank you for your patience with this, I’ve tried several approaches, including explicitly enabling Corepack and manually installing Yarn, but I’m still running into issues with getting Yarn v4.3.1 to work in the GitHub Actions workflow.
I would appreciate any guidance or suggestions you may have to help resolve this.

vault-developer commented:
> Hello @vault-developer Thank you for your patience with this, I’ve tried several approaches, including explicitly enabling Corepack and manually installing Yarn, but I’m still running into issues with getting Yarn v4.3.1 to work in the GitHub Actions workflow. I would appreciate any guidance or suggestions you may have to help resolve this.

Hey @iampearlep no worries at all 👍 
We already have a working example of installing dependencies and enabling core pack in another GA:
https://github.com/vault-developer/event-loop-explorer/actions/runs/11193111306/workflow
Could you check please?

vault-developer commented:
Hey @iampearlep I made some changes for eslint, it should not throw errors anymore 👍

iampearlep commented:
> Hey @iampearlep I made some changes for eslint on master branch, it should not throw errors anymore 👍

Okay, thank you.

iampearlep commented:
Hello @vault-developer 
I have tested it and it works fine now, thank you.

vault-developer commented:
Thank you for the pr!
